apps/212-shinymeta/shinyjster.js

```diff
@@ -11,23 +11,38 @@ function selectOption(jst, id, index) {
 var jst = jster();
 jst.add(Jster.shiny.waitUntilStable);
 
-selectOption(jst, "x-col", 1);
-selectOption(jst, "y-col", 2);
-jst.add(Jster.shiny.waitUntilStable);
-jst.add(function() {
-  var btn = $("button");
-  if (btn.length !== 1) {
-    throw new Error("One and only one button was expected, found " + btn.length);
-  }
-  btn.click();
-});
+selectOption(jst, "x-col", 1); // choose cyl
+selectOption(jst, "y-col", 2); // choose disp
+
 jst.add(Jster.shiny.waitUntilStable);
-jst.add(function() {
-  var lines = $(".shiny-ace").data("aceEditor").session.doc.getAllLines();
-  Jster.assert.isEqual(
-    lines.join(" \n "),
-    "dataset <- mtcars \n x_values <- dataset %>% \n   dplyr::pull(!!as.symbol(\"cyl\")) \n y_values <- dataset %>% \n   dplyr::pull(!!as.symbol(\"disp\")) \n # Combine x and y into data frame for plotting \n df_plot <- data.frame(x = x_values, y = y_values) \n plot(df_plot) \n x_avg <- x_values %>% \n   mean() %>% \n   round(1) \n paste(\"Average of\", \"cyl\", \"is\", x_avg) \n y_avg <- y_values %>% \n   mean() %>% \n   round(1) \n paste(\"Average of\", \"disp\", \"is\", y_avg)"
-  );
+
+// renderPlot() should be 1st, plotly 2nd, DT 3rd
+var expected_cases = [
+  "plot(df_plot)",
+  "plot_ly(df_plot, x = ~x, y = ~y) %>% \n   add_markers()",
+  "datatable(df_plot)"
+];
+
+function expected_script(case_idx) {
+    return "dataset <- mtcars \n x_values <- dataset %>% \n   dplyr::pull(!!as.symbol(\"cyl\")) \n y_values <- dataset %>% \n   dplyr::pull(!!as.symbol(\"disp\")) \n # Combine x and y into data frame for plotting \n df_plot <- data.frame(x = x_values, y = y_values) \n " +
+    expected_cases[case_idx] +
+    " \n x_avg <- x_values %>% \n   mean() %>% \n   round(1) \n paste(\"Average of\", \"cyl\", \"is\", x_avg) \n y_avg <- y_values %>% \n   mean() %>% \n   round(1) \n paste(\"Average of\", \"disp\", \"is\", y_avg)";
+}
+
+$("button").each(function(idx) {
+  var btn = $(this);
+  jst.add(function() {
+    btn.click();
+  });
+  jst.add(Jster.shiny.waitUntilStable);
+  jst.add(function() {
+    var lines = $(".shiny-ace").data("aceEditor").session.doc.getAllLines();
```

Comment:
This line is failing on Windows. It seems like the data entry being returned is not working. As if the modal is not being displayed.

I'm ok with disabling the tests for Windows. (The waitUntilStable seems like it should be correct to wait for the modal text to finish calculating within Shiny and display on the window.)